src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/UnusedLocalVariableCheck.java

```diff
@@ -512,7 +512,7 @@ private void addInstanceOrClassVar(DetailAST varDefAst) {
                 && !isPrivateInstanceVariable(varDefAst)) {
             final DetailAST ident = varDefAst.findFirstToken(TokenTypes.IDENT);
             final VariableDesc desc = new VariableDesc(ident.getText(),
-                    varDefAst.findFirstToken(TokenTypes.TYPE), findScopeOfVariable(varDefAst));
+                    null, null);
```

Comment:
I too analyzed the code and yes, changing it to null won't have any effect because:

1. `DetailAST typeAst` is only used while logging violations, and we only log violations for local variables, and this `VariableDesc` is for an instance or class variable.
2. `DetailAST scope`. We never use the initially set scope. When we use `scope`, we update it to our value before using it. See https://github.com/checkstyle/checkstyle/blob/3f77dd11d7e7ac8e23ecc817a1caeeea7d1a489d/src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/UnusedLocalVariableCheck.java#L1010-L1020 we use `updatedScope` instead.

So all in all, we can safely do these changes BUT this is not a good design. The design of the check should be altered to not have situations like these. This is out of the scope of this summer's project, a separate issue to enhance the design of the check should be created.

---

LGTM!

Reply:
Ok, I don't have time now to investigate in detail.

We will deal with this in separate project